contrib/olivere/elastic/elastictrace.go

```diff
@@ -1,86 +1,120 @@
-// Package elastictrace provides tracing for the Elastic Elasticsearch client.
-// Supports v3 (gopkg.in/olivere/elastic.v3), v5 (gopkg.in/olivere/elastic.v5)
-// but with v3 you must use `DoC` on all requests to capture the request context.
-package elastictrace
+// Package elastic provides functions to trace the gopkg.in/olivere/elastic.v{3,5} packages.
+package elastic
 
 import (
 	"bytes"
 	"errors"
+	"fmt"
 	"io/ioutil"
+	"net"
 	"net/http"
+	"regexp"
 	"strconv"
+	"time"
 
 	"github.com/DataDog/dd-trace-go/tracer"
 	"github.com/DataDog/dd-trace-go/tracer/ext"
 )
 
-// MaxContentLength is the maximum content length for which we'll read and capture
-// the contents of the request body. Anything larger will still be traced but the
-// body will not be captured as trace metadata.
-const MaxContentLength = 500 * 1024
+// NewHTTPClient returns a new http.Client which traces requests under the given service name.
+//
+// TODO(gbbr): Remove tracer argument when we switch to OpenTracing.
+func NewHTTPClient(service string, tracer *tracer.Tracer) *http.Client {
+	return &http.Client{Transport: &httpTransport{&http.Transport{
+		// http.DefaultTransport
+		Proxy: http.ProxyFromEnvironment,
+		DialContext: (&net.Dialer{
+			Timeout:   30 * time.Second,
+			KeepAlive: 30 * time.Second,
+			DualStack: true,
+		}).DialContext,
+		MaxIdleConns:          100,
+		IdleConnTimeout:       90 * time.Second,
+		TLSHandshakeTimeout:   10 * time.Second,
+		ExpectContinueTimeout: 1 * time.Second,
+	}, tracer, service}}
+}
 
-// TracedTransport is a traced HTTP transport that captures Elasticsearch spans.
-type TracedTransport struct {
-	service string
-	tracer  *tracer.Tracer
+// NewHTTPClientWithTransport returns a new http.Client that traces requests using
+// the given http.Transport.
+//
+// TODO(gbbr): Remove tracer argument when we switch to OpenTracing.
+func NewHTTPClientWithTransport(service string, tracer *tracer.Tracer, transport *http.Transport) *http.Client {
```

Comment:
shouldn't the `transport` be the first argument? just to keep our arguments as last (like we did in other integrations)